sentry-android-core/src/main/java/io/sentry/android/core/EnvelopeFileObserver.java

```diff
@@ -0,0 +1,39 @@
+package io.sentry.android.core;
+
+import android.os.FileObserver;
+import io.sentry.core.*;
+import io.sentry.core.util.Objects;
+
+final class EnvelopeFileObserver extends FileObserver {
```

Comment:
Java has the `WatchService` but it's only available API >= 26 for Android.
https://developer.android.com/reference/java/nio/file/WatchService

Reply:
for `FileObserver` pay attention to that:

> Warning: If a FileObserver is garbage collected, it will stop sending events. To ensure you keep receiving events, you must keep a reference to the FileObserver instance from some other live object.

Reply:
there are backports of `WatchService` for Android, like https://github.com/lukhnos/nnio but not sure how trustable it is

Reply:
apache commons-vfs also has classes for watching a dir.

Reply:
It won't be GC'ed because the reference is held by the integration.
Why use `WatchService` over `FileObserver`?

Reply:
if it would be possible `WatchService` is pure-java, so not dependent on the Android FW and could be on sentry-core, but it's min. API 26 so :(

Reply:
Makes sense. For now let's target only `sentry-android` which requires API higher than 26 so I'd say `FileObserver` should be OK. After the 25th we can put all options back on the table. 
Once backend java becomes priority this will come up anyway. In fact, since there's no NDK there we might just not have this at all. Only the ReadOnStartup behavior

Reply:
agreed